plugins/optimization-detective/storage/class-od-rest-url-metrics-store-endpoint.php

```diff
@@ -0,0 +1,364 @@
+<?php
+/**
+ * REST API integration for the plugin: OD_REST_URL_Metrics_Store_Endpoint.
+ *
+ * @package optimization-detective
+ * @since 0.1.0
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * OD_REST_URL_Metrics_Store_Endpoint class
+ *
+ * @since n.e.x.t
+ *
+ * @access private
+ */
+class OD_REST_URL_Metrics_Store_Endpoint {
+
+	/**
+	 * Get the namespace for optimization-detective
+	 *
+	 * @since n.e.x.t
+	 * @return non-empty-string Namespace.
+	 */
+	public static function get_namespace(): string {
+		return OD_REST_API_NAMESPACE; // @phpstan-ignore constant.deprecated (To be replaced with string literal when constant is removed.)
+	}
+
+	/**
+	 * Gets the route for storing a URL Metric.
+	 *
+	 * Note the `:store` art of the endpoint follows Google's guidance in AIP-136 for the use of the POST method in a way
+	 * that does not strictly follow the standard usage. Namely, submitting a POST request to this endpoint will either
+	 * create a new `od_url_metrics` post, or it will update an existing post if one already exists for the provided slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @link https://google.aip.dev/136
+	 *
+	 * @return non-empty-string Route.
+	 */
+	public static function get_route(): string {
```

Comment:
Same here, I think this would be better a `const` on the class.

Reply:
See https://github.com/WordPress/performance/pull/1865#discussion_r1990209610